rust/bridge/node/futures/src/exception.rs

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright 2021 Signal Messenger, LLC.
+// SPDX-License-Identifier: AGPL-3.0-only
+//
+
+use neon::prelude::*;
+
+/// See [PersistentException]
+enum PersistentExceptionImpl {
+    Object(Root<JsObject>),
+    String(String),
+}
+
+/// Persists a JavaScript exception across call contexts for later re-throwing
```

Comment:
I considered a more generic name, but ultimately went with the more limited thing so that the library had an opinionated view of the type and how it was to be used. (It shows up in Results, for example, which doesn't quite make sense for arbitrary persisted JS values.) I hadn't fully considered the similarity to ManuallyDrop, but I think matching Neon's Root is more important.